### ISSUE TITLE
Deploy to all cloudbank hubs can run into rate limitations

### ISSUE THREAD

**Issue by consideRatio** [2i2c-org/infrastructure#4379]
When the deployer does the helm upgrade command, it can lead to the following error when doing it to soo many cloudbank hubs.

```shell
Error: UPGRADE FAILED: client rate limiter Wait returned an error: context deadline exceeded
```

**Comment by sgibson91** [comment]
We could limit the number of concurrent jobs? https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-a-matrix-for-your-jobs#defining-the-maximum-number-of-concurrent-jobs

**Comment by consideRatio** [comment]
Yeah that would help @sgibson91, and would likely be without compromise if we could limit the number of parallel jobs to a specific cluster - as we aren't rate limited when working against many separate clusters' hubs. It would slow down our CD system notably if we couldn't constrain max per cluster though.

**Comment by consideRatio** [comment]
I think https://github.com/2i2c-org/infrastructure/issues/3222 could be related. By a change there, we reduce the amount of requests.
